pkg/proxy/iptables/proxier.go

```diff
@@ -1623,6 +1613,25 @@ func (proxier *Proxier) syncProxyRules() {
 			v.Close()
 		}
 	}
+
+	// Clear conntrack entries for the new UDP NodePorts, this has to be done AFTER the iptables rules are programmed.
+	// It can happen that traffic to the NodePort hits the host before the iptables rules are programmed
+	// this will create an stale entry in conntrack that will blackhole the traffic, so we need to clear it
```

Comment:
This opens up the opposite race condition: now if someone connects to the service in between when we program iptables and here, that will create a conntrack entry for the DNAT mapping, but then we'll delete that entry here, which means the reply packets will get lost.

Do we _only_ want to delete blackhole entries? If so, can we pass an argument to `conntrack` specifying that? (`-u <something>` maybe?)

Maybe since this is UDP we don't care; we're already claiming below that accidentally clearing out the wrong conntrack entries is "very low impact".

Reply:
yeah, I was testing this patch and it keeps failing, I think that I'm hitting exactly what you are describing ...

> Do we only want to delete blackhole entries? If so, can we pass an argument to conntrack specifying that? (-u <something> maybe?)

NodePort can receive from any ip and forward to any endpoint , can you elaborate?

Reply:
I mean is there some way we can say "delete entries that say 'this IP doesn't exit' but don't delete entries that say 'this IP _does_ exist'"

Reply:
maybe if we can delete only the UNREPLIED?
```
udp      17 28 src=192.168.1.150 dst=192.168.1.144 sport=12345 dport=31123 [UNREPLIED] src=192.168.1.144 dst=192.168.1.150 sport=31123 dport=12345 mark=0 use=1
```

conntrack-tools has an status flag but I´m not able to make it work with that status
```
  -u, --status status           Set status, eg. ASSURED
```

Reply:
hmm it is not the reply thing :thinking: ... let me try something

Reply:
this is solved now